.travis.yml

```diff
@@ -18,74 +16,57 @@ env:
+    - if [ "$CXX" = "g++" ]; then export CXX="g++-6"; export CC="gcc-6"; export NO_ERROR_ON_MAYBE_UNINITIALIZED="-Wno-error=maybe-uninitialized"; fi
 
     # Clang
-    - if [ "$CXX" == "clang++" ]; then sudo apt-get install --allow-unauthenticated -qq clang-3.8; export CXX="clang++-3.8"; export CC="clang-3.8"; fi
+    - if [ "$CXX" == "clang++" ]; then export CXX="clang++-3.8"; export CC="clang-3.8"; fi
 
     # Don't Wno-error-maybe-uninitialized if we're in a Debug build
     - if [ "$CONFIG" == "Debug"]; then export NO_ERROR_ON_MAYBE_UNINITIALIZED=""; fi
 
     # CMake
     - tar -zxf ${CMAKE_VERSION}.tar.gz -C build
 
-    # Install MongoDB Enterprise
-    - sudo apt-get install mongodb-enterprise-server
-
-    # Install CMake (Versions lower than 2.8.8 do not support object libraries)
-    - sudo apt-get install cmake
-
     # Prepare pkg-config path
-    - export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:/home/travis/install/lib/pkgconfig/
-
-    # Install Mongo C Driver
-    - tar -zxf mongo-c-driver-${CDRIVER_VERSION}.tar.gz
-    - pushd mongo-c-driver-${CDRIVER_VERSION}; ./configure --prefix=/home/travis/install; make; sudo make install; popd;
+    - export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:$HOME/deps/lib/pkgconfig/
 
-    # Install Mongo C++ Driver
-    - pushd mongo-cxx-driver/build; ../../build/${CMAKE_BINARY} -DCMAKE_CXX_FLAGS="-Wno-ignored-attributes" -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/home/travis/install/ ..; make; sudo make install; popd; 
+    # Install dependencies
+    - git clone -b 'master' --single-branch https://github.com/mongodb/mongo-cxx-driver.git
+    - bash travis-install-deps.sh
 
 before_script:
     - $CC --version
     - $CXX --version
 
     # Build the odm and the tests
+    - export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:$HOME/install/lib/pkgconfig
     - cd build
-    - ${CMAKE_BINARY} -DCMAKE_BUILD_TYPE=$CONFIG -DCMAKE_MODULE_PATH=/home/travis/build/mongodb/mongo-cxx-odm/mongo-cxx-driver/cmake -DCMAKE_INSTALL_PREFIX=/home/travis/install -DCMAKE_C_FLAGS="-Wall -Wextra -Wno-attributes -Werror -Wno-error=missing-field-initializers ${NO_ERROR_ON_MAYBE_UNINITIALIZED}" -DCMAKE_CXX_FLAGS="-Wall -Wextra -Wno-attributes -Werror -Wno-error=missing-field-initializers ${NO_ERROR_ON_MAYBE_UNINITIALIZED}" ..
+    - ${CMAKE_BINARY} -DCMAKE_BUILD_TYPE=$CONFIG -DCMAKE_MODULE_PATH=$HOME/build/mongodb/mongo-cxx-odm/mongo-cxx-driver/cmake -Dlibbsoncxx_DIR=$HOME/deps/lib/cmake/libbsoncxx-3.0.2-rc0-pre -Dlibmongocxx_DIR=$HOME/deps/lib/cmake/libmongocxx-3.0.2-rc0-pre -DCMAKE_INSTALL_PREFIX=$HOME/install -DCMAKE_C_FLAGS="-Wall -Wextra -Wno-attributes -Werror -Wno-error=missing-field-initializers ${NO_ERROR_ON_MAYBE_UNINITIALIZED}" -DCMAKE_CXX_FLAGS="-Wall -Wextra -Wno-attributes -Werror -Wno-error=missing-field-initializers ${NO_ERROR_ON_MAYBE_UNINITIALIZED}" ..
```

Comment:
Seems like `CMAKE_C_FLAGS` and `CMAKE_CXX_FLAGS` are the same? I think cmake might support `CMAKE_CC_FLAGS` to send to both. If not, can you create a bash variable to pass to these?